zerver/tests/test_signup.py

```diff
@@ -1720,6 +1720,37 @@ def test_accessing_invites_in_another_realm(self) -> None:
         error_result = self.client_delete('/json/invites/' + str(prereg_user.id))
         self.assert_json_error(error_result, "No such invitation")
 
+    def test_prereg_user_status(self) -> None:
+        email = self.nonreg_email("alice")
+        password = "password"
+        realm = get_realm('zulip')
+
+        inviter = UserProfile.objects.first()
+        prereg_user = PreregistrationUser.objects.create(
+            email=email, referred_by=inviter, realm=realm)
+
+        confirmation_link = create_confirmation_link(prereg_user, 'host', Confirmation.USER_REGISTRATION)
+        registration_key = confirmation_link.split('/')[-1]
+
+        result = self.client_post(
+            "/accounts/register/",
+            {"key": registration_key,
+             "from_confirmation": "1",
+             "full_name": "alice"})
+        confirmation = Confirmation.objects.get(confirmation_key=registration_key)
+        prereg_user = confirmation.content_object
+        self.assertEqual(prereg_user.status, 0)
+        self.assertEqual(result.status_code, 200)
```

Comment:
```suggestion
        self.assertEqual(result.status_code, 200)
        confirmation = Confirmation.objects.get(confirmation_key=registration_key)
        prereg_user = confirmation.content_object
        self.assertEqual(prereg_user.status, 0)
```
This is a nitpick, but I think it generally makes the most sense to check the http result first? Since that'll give the clearest test failure error if the status code becomes something unexpected.